.github/workflows/build.yaml

```diff
@@ -22,29 +24,72 @@ jobs:
       matrix:
         include:
         - { arch: aarch64, runs-on: macos-14 }
-        - { arch: x86_64, runs-on: macos-13 }
+        - { arch: x86_64, runs-on: macos-12 }
     runs-on: ${{ matrix.runs-on }}
-    timeout-minutes: 40
+    timeout-minutes: 120
     steps:
-    - run: mkdir -p build
-    - run: brew install filemonitor ninja
-    - run: brew install --ignore-dependencies lima # Don't pull in qemu
     - uses: actions/checkout@v4
       with:
         persist-credentials: false
-    - run: brew install --build-from-source ./qemu.rb
+        fetch-depth: 0
+    - id: version
+      name: Calculate version
+      shell: bash
+      run: |
+        version=$(git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
+        echo "VERSION=${version#v}" >>"$GITHUB_OUTPUT"
+    - name: Uninstall all brew packages and clear cache
+      run: |
+        set -o xtrace
+        brew list | xargs brew uninstall --force
+        brew autoremove
+        brew cleanup
+        rm -rf "$(brew --cache)"
+    - name: Build qemu and all prerequisites from source
+      run: |
+        set -o xtrace
+        brew tap-new rd/tap
+        cp qemu.rb "$(brew --repo rd/tap)/Formula/qemu.rb"
```

Comment:
Would it make sense to checkout into the directory instead?

Reply:
I'm not sure I understand the question. This step is creating a new tap, that will be inside the homebrew-managed directory structure. Do you want to create a separate repo to store a tap on GitHub? That seems like overkill compared to copying a single file.

Reply:
We have a `actions/checkout` step, and here we copy the `qemu.rb` into the tap directory. I was wondering if calling `actions/checkout` directly into `…/Formula` would be useful.

Reply:
But then everything else from this repo would be inside the new tap too. And directory doesn't even exist yet. I don't know how this would be simpler; it feels more complicated, and having to call the `lima-and-qemu.pl` script like this looks just wrong to me:

```
perl "$(brew --repo rd/tap)/Formula/lima-and-qemu.pl"
```

I'm sorry, I must be missing something.